### ISSUE TITLE
GRPC Builds Fail by default on CentOS 7

### ISSUE THREAD

**Issue by loganasherjones** [rules-proto-grpc/rules_proto_grpc#231]
### Issue Description

Thanks for the wonderful tool. I'm on a CentOS 7 machine and cannot build `example/go/go_grpc_compile`. Here are the steps to reproduce on a CentOS 7 machine:

```
git clone https://github.com/rules-proto-grpc/rules_proto_grpc
cd rules_proto_grpc
git checkout 4.2.0
cd example/go/go_grpc_compile
bazel build :all
```

The problem appears to be with the `//tools/fixer:fixer` binary. It uses `std::regex` and the default `gcc` on CentOS 7 is an older version of gcc that doesn't support regex.

I can work around this by installing a newer version of gcc and then configuring the `CC` environment variable, but as far as I can tell, the only reason I need to do this is for the `fixer` binary. Since I'm not using C++ in my project, it feels odd to have to configure `gcc`. It would be nice if I didn't have to do this.

I'm not really a big C++ person so figuring out what went wrong was not very straightforward. It would be nice if something failed early and told me I'm on a too old version of GCC or if perhaps fixer was re-written either in a different language or to not require `std::regex`.

### Log Output

```plain text
INFO: Analyzed 2 targets (0 packages loaded, 0 targets configured).
INFO: Found 2 targets...
ERROR: /home/logan/workspace/rules_proto_grpc/example/go/go_grpc_compile/BUILD.bazel:8:16: Applying fixer for grpc_go_plugin plugin on target //:greeter_go_grpc failed: (Aborted): process-wrapper failed: error executing command 
  (cd /home/logan/.cache/bazel/_bazel_docker/cef2209ae0d86490bf5b76c183c648ac/sandbox/processwrapper-sandbox/232/execroot/__main__ && \
  exec env - \
    TMPDIR=/tmp \
  /home/logan/.cache/bazel/_bazel_docker/install/cbf972266931ad9fad1857441b832915/process-wrapper '--timeout=0' '--kill_delay=15' bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_proto_grpc/tools/fixer/fixer bazel-out/k8-fastbuild/bin/_rpg_premerge_greeter_go_grpc/_plugin_fixer_manifest_grpc_go_plugin.txt external/rules_proto_grpc/go/empty.template bazel-out/k8-fastbuild/bin/_rpg_premerge_greeter_go_grpc/_plugin_fixed_grpc_go_plugin bazel-out/k8-fastbuild/bin/_rpg_premerge_greeter_go_grpc)
terminate called after throwing an instance of 'std::regex_error'
  what():  regex_error
INFO: Elapsed time: 4.342s, Critical Path: 0.20s
INFO: 3 processes: 3 internal.
FAILED: Build did NOT complete successfully
```
```


### rules_proto_grpc Version

4.2.0

### Bazel Version

5.3.2

### OS

CentOS 7

### Link to Demo Repo

_No response_

### WORKSPACE Content

```Starlark
See `example/go/go_grpc_compile/WORKSPACE`
```

### BUILD Content

```Starlark
See `example/go/go_grpc_compile/BUILD.bazel`
```

**Comment by loganasherjones** [comment]
Thanks for the response and explanation. I understand this might not be important enough to fix, I just wanted to make sure I documented it at least somewhere.

I tried with the options you suggested and it still complains about the same problem. The problem is that the default `gcc` version is `4.8.5` which doesn't have support for regex. I think support was added in `4.9`. Just kinda stinks because now I've got to figure out how to configure a cc toolchain which I'm totally qualified to do :sweat_smile:
